subdoc/lib/run.cc

```diff
@@ -80,13 +79,15 @@ sus::Result<Database, DiagnosticResults> run_files(
       std::make_shared<clang::PCHContainerOperations>(), std::move(fs));
   tool.setDiagnosticConsumer(&*diags);
 
-  auto adj = [](clang::tooling::CommandLineArguments args, llvm::StringRef) {
+  ClangResourceDir resource_dir;
+
+  auto adj = [&resource_dir](clang::tooling::CommandLineArguments args,
+                             llvm::StringRef) {
     // Clang-cl doesn't understand this argument, but it may appear in the
     // command-line for MSVC in C++20 codebases (like subspace).
     std::erase(args, "/Zc:preprocessor");
 
-    const std::string& tool = args[0];
-    if (tool.find("cl.exe") != std::string::npos) {
+    if (args[0].find("cl.exe") != std::string::npos) {
```

Comment:
probably this should also use filename extraction?

Reply:
Hmm yeah it should :) That'd be more robust.